x-pack/plugins/observability/public/plugin.ts

```diff
@@ -242,6 +252,8 @@ export class Plugin
       navigation: {
         registerSections: this.navigationRegistry.registerSections,
       },
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      useRulesLink: useRulesLinkCreator(config.unsafe.rules.enabled),
```

Comment:
Maybe if you renamed the function to `createUseRulesLink` you could avoid the lint issues.

The current name implies that calling `useRulesLinkCreator` will give you back a `RulesLinkCreator` to me.

Reply:
@miltonhultgren I changed the name to what you suggested and linter is happy!

Reply:
How about file name? Would you keep it as it is so that it starts with the use prefix as rest hooks in the folder `use_rules_link_creator` or rename to `create_use_rules_link` to match the name of the function?

Reply:
I, personally, would rename the file as well. 
It's just to make it clear to the person scrolling those files that there is something slightly different about this file, since it exposes a "hook creator" rather than a hook (like the other files in that folder).
But I think in the big picture that might be a detail.

Reply:
Ok I agree! Will change the filename as well